lib/Auth/Process/UpdateAUP.php

```diff
@@ -59,15 +59,23 @@ public function __construct($config, $reserved)
     public function process(&$state)
     {
         if (!empty($this->config['spBlacklist']) && isset($state['SPMetadata']['entityid']) && in_array($state['SPMetadata']['entityid'], $this->config['spBlacklist'], true)) {
-            SimpleSAML_Logger::debug("[aup] process: Skipping blacklisted SP ". var_export($state['SPMetadata']['entityid'], true));
+            SimpleSAML_Logger::debug("[aup:UpdateAUP] Skipping blacklisted SP ". var_export($state['SPMetadata']['entityid'], true));
             return;
         }
         // Check if user is in blacklist
-        if(!empty($this->config['userIdAttribute']) && !empty($this->config['userIdBlacklist']) &&  !empty(array_intersect($state['Attributes'][$this->config['userIdAttribute']], $this->config['userIdBlacklist']))) {
+        if ((!empty($this->config['userIdAttribute']) && !empty($this->config['userIdBlacklist']) && !empty(array_intersect(
+            $state['Attributes'][$this->config['userIdAttribute']],
+            $this->config['userIdBlacklist'])))) {
+            SimpleSAML_Logger::debug("[aup:UpdateAUP] Skipping blacklisted user with id " . $state['Attributes'][$this->config['userIdAttribute']]);
```

Comment:
this is an array so better use `var_export($state['Attributes'][$this->config['userIdAttribute'], true)`

Reply:
done